src/CentrifugeRouter.sol

```diff
@@ -104,7 +113,10 @@ contract CentrifugeRouter is Auth, ICentrifugeRouter {
 
     /// @inheritdoc ICentrifugeRouter
     function claimDeposit(address vault, address receiver, address controller) external protected {
-        require(controller == _initiator || controller == receiver, "CentrifugeRouter/invalid-sender");
+        require(
+            controller == _initiator || (controller == receiver && opened[controller][vault] == true),
+            "CentrifugeRouter/invalid-sender"
```

Comment:
I think you shouldn't  rely on  the controller to call `close` externally but rather  call it here. Otherwise first `claimDeposit` for this controller on this vault is protected but any future operations might be susceptible to the same issue.
Same goes for `claimRedeem`